### DISCUSSION TITLE
Searching Relationships doesn't work properly in dot-notation

### DISCUSSION THREAD

**Discussion by TarasZakus** [laravel/nova-issues#4168] [Ideas & Feature Requests]
- Laravel Version: 8.83.10
- Nova Version: 4.2.7
- PHP Version: 8.0.18
- Database Driver & Version: pgsql 13.5
- Operating System and Version: Ubuntu 20.04.4
- Browser type and version: doesn't matter.
- Reproduction Repository: https://github.com/laravel/nova

### Description:
[Nova Documentation](https://nova.laravel.com/docs/4.0/search/#searching-relationships) says:
> you may define a relationship that should be searched by adding the field to your resource's $search property using "dot notation"

So, for example:
```php
public static $search = [
    'id', // works
    'author.name', // works
    'author.mother.name', // doesn't work
];
```
If we define a relationship that is more than 1 level deep, then it produces a DB error like this:
`ERROR:  missing FROM-clause entry for table "mother" at character...`

### How to fix:
Consider changing this line:
```php
return (new Column($this->column))->__invoke(
```
to this:
```php
return (Column::from($this->column))->__invoke(
```
in the following methods:
```
\Laravel\Nova\Query\Search\SearchableRelation::__invoke()
\Laravel\Nova\Query\Search\SearchableMorphToRelation::__invoke()
```
and the issue will be resolved.

**Comment by davidhemphill** [comment]
There was a pull request merged just adding this. It should be in the next release. 👍
